src/solvers/dg/1d/containers.jl

```diff
@@ -32,41 +65,72 @@ function ElementContainer1D{RealT, NVARS, POLYDEG}(capacity::Integer) where {Rea
   nan = convert(RealT, NaN)
 
   # Initialize fields with defaults
-  u = fill(nan, NVARS, n_nodes, capacity)
-  u_t = fill(nan, NVARS, n_nodes, capacity)
+  _u = fill(nan, NVARS * n_nodes * capacity)
+  u = unsafe_wrap(Array, pointer(_u),
+                  (NVARS, n_nodes, capacity))
+
+  _u_t = fill(nan, NVARS *  n_nodes * capacity)
+  u_t = unsafe_wrap(Array, pointer(_u_t),
+                    (NVARS, n_nodes, capacity))
+
   # u_rungakutta is initialized to non-NaN since it is used directly
-  u_tmp2 = fill(zero(RealT), NVARS, n_nodes, capacity)
-  u_tmp3 = fill(zero(RealT), NVARS, n_nodes, capacity)
+  _u_tmp2 = fill(zero(RealT), NVARS * n_nodes * capacity)
+  u_tmp2 = unsafe_wrap(Array, pointer(_u_tmp2),
+                       (NVARS, n_nodes, capacity))
+
+  _u_tmp3 = fill(zero(RealT), NVARS * n_nodes * capacity)
+  u_tmp3 = unsafe_wrap(Array, pointer(_u_tmp3),
+                       (NVARS, n_nodes, capacity))
+
   inverse_jacobian = fill(nan, capacity)
-  node_coordinates = fill(nan, 1, n_nodes, capacity)
-  surface_flux_values = fill(nan, NVARS, 2 * 1, capacity)
+
+  _node_coordinates = fill(nan, 1 * n_nodes * capacity)
+  node_coordinates = unsafe_wrap(Array, pointer(_node_coordinates),
+                                 (1, n_nodes, capacity))
+
+  _surface_flux_values = fill(nan, NVARS * 2 * 1 * capacity)
+  surface_flux_values = unsafe_wrap(Array, pointer(_surface_flux_values),
+                                    (NVARS, 2 * 1, capacity))
+
   cell_ids = fill(typemin(Int), capacity)
 
-  elements = ElementContainer1D{RealT, NVARS, POLYDEG}(
+  return ElementContainer1D{RealT, NVARS, POLYDEG}(
     u, u_t, u_tmp2, u_tmp3,
-    inverse_jacobian, node_coordinates, surface_flux_values, cell_ids)
-
-  return elements
+    inverse_jacobian, node_coordinates, surface_flux_values, cell_ids,
+    _u, _u_t, _u_tmp2, _u_tmp3, _node_coordinates, _surface_flux_values)
 end
 
 
 # Return number of elements
-nelements(elements::ElementContainer1D) = length(elements.cell_ids)
+@inline nelements(elements::ElementContainer1D) = length(elements.cell_ids)
 
 
 # Container data structure (structure-of-arrays style) for DG interfaces
 # TODO: Taal refactor, remove NVARS, POLYDEG?
-# TODO: Taal refactor, mutable struct or resize! for AMR?
 mutable struct InterfaceContainer1D{RealT<:Real, NVARS, POLYDEG} <: AbstractContainer
   u::Array{RealT, 3}        # [leftright, variables, interfaces]
   neighbor_ids::Matrix{Int} # [leftright, interfaces]
   orientations::Vector{Int} # [interfaces]
+  # internal `resize!`able storage
+  _u::Vector{RealT}
+  _neighbor_ids::Vector{Int}
 end
 
-function Base.copy!(dst::InterfaceContainer1D, src::InterfaceContainer1D)
-  dst.u            = src.u
-  dst.neighbor_ids = src.neighbor_ids
-  dst.orientations = src.orientations
+function Base.resize!(interfaces::InterfaceContainer1D{RealT, NVARS, POLYDEG},
```

Comment:
Same argument here about a (short) comment explaining things like "internal storage" and why "unsafe wrapping" is used.

Reply:
Done in 18e3930

Reply:
Looks good, thanks!